### PR TITLE
peppermint changes

### PR DESCRIPTION
**Pull request by jdkanani**

### CODE REVIEW

**Comment by jdkanani** [comment]
@vaibhavchellani pushed new changes

File: consensus/state.go [hunk 1]
```diff
@@ -1671,7 +1683,14 @@ func (cs *ConsensusState) signVote(type_ byte, hash []byte, header types.PartSet
...
 		BlockID:          types.BlockID{hash, header},
 	}
+
+	if cs.LockedBlock != nil {
+		cs.Logger.Info("[peppermint] sign add vote", "lockedHeaderHash", hex.EncodeToString(header.Hash), "lockedBlockDataHash", hex.EncodeToString(cs.LockedBlock.DataHash))
+		vote.Data = cs.LockedBlock.DataHash
+	}
+
 	err := cs.privValidator.SignVote(cs.state.ChainID, vote)
+	cs.Logger.Info("[peppermint] vote sign with data", "signBytes", vote.SignBytes(cs.state.ChainID))
```
Comment (jdkanani) [inline consensus/state.go]: @vaibhavchellani here we will have now signed vote with tx hash.

**Comment by jdkanani** [comment]
merging to master for now.

File: consensus/state.go [hunk 2]
```diff
@@ -1290,6 +1297,11 @@ func (cs *ConsensusState) finalizeCommit(height int64) {
 		// but may differ from the LastCommit included in the next block
 		precommits := cs.Votes.Precommits(cs.CommitRound)
 		seenCommit := precommits.MakeCommit()
+		precommitStrings := make([]string, len(seenCommit.Precommits))
```
Comment (vaibhavchellani) [inline consensus/state.go]: what is the user of this array ?

Comment (jdkanani) [reply consensus/state.go]: Added it just for the log. You can remove it if you know where to look for vote sigs.

**Pull request merged**